From cf3b47e856ae7bd7cc2771677d5896d79931fac1 Mon Sep 17 00:00:00 2001
From: Luc Verhaegen <libv@codethink.co.uk>
Date: Thu, 25 Oct 2012 04:04:22 +0200
Subject: [PATCH 890/944] video:sunxi:lcd: fix remaining sparse issues

* warning: non-ANSI function declaration of function
  'check_spi_used_value'
* warning: Using plain integer as NULL pointer

Signed-off-by: Luc Verhaegen <libv@codethink.co.uk>
---
 drivers/video/sunxi/lcd/lcd_panel_cfg_generic.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/video/sunxi/lcd/lcd_panel_cfg_generic.c b/drivers/video/sunxi/lcd/lcd_panel_cfg_generic.c
index 3e31ca5..5c89820 100644
--- a/drivers/video/sunxi/lcd/lcd_panel_cfg_generic.c
+++ b/drivers/video/sunxi/lcd/lcd_panel_cfg_generic.c
@@ -38,7 +38,8 @@ static __s32 lcd_spi_mosi;
 static __s32 lcd_spi_used;
 static __s32 lcd_spi_module = -1;
 
-static void check_spi_used_value()
+static void
+check_spi_used_value(void)
 {
 	if (SCRIPT_PARSER_OK !=
 	    OSAL_Script_FetchParser_Data("lcd_spi_para", "lcd_spi_used",
@@ -264,7 +265,8 @@ __s32 LCD_user_defined_func_generic(__u32 sel, __u32 para1, __u32 para2,
 
 void LCD_get_panel_funs_generic(__lcd_panel_fun_t *fun)
 {
-	fun->cfg_panel_info = 0; /* lcd panel info defined in sys_config1.fex */
+	/* lcd panel info defined in sys_config1.fex */
+	fun->cfg_panel_info = NULL;
 	fun->cfg_open_flow = LCD_open_flow_generic;
 	fun->cfg_close_flow = LCD_close_flow_generic;
 	fun->lcd_user_defined_func = LCD_user_defined_func_generic;
-- 
1.8.0

